### PR TITLE
Aad role setting

### LINKED ISSUE #2227: AADRoleSetting throws an error if there are no AAD P2 licenses in a tenant
#### Details of the scenario you tried and the problem that is occurring
Call Export-M365DSCConfiguration -Workload AAD
throws an error when attempting to export AADRoleSettings

#### Verbose logs showing the problem
[8/11] Extracting [AADRoleSetting]...
    |---[1/92] Global AdministratorGet-MgPolicyRoleManagementPolicyAssignment : The tenant needs an AAD Premium 2 licens
e.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.824.1\DSCResources\MSFT_AADRoleSetting\MSFT_AADRoleS
etting.psm1:215 char:5
+     $Policy = Get-MgPolicyRoleManagementPolicyAssignment -Filter $Fil ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: ({ Top = , Skip ...ndProperty =  }:<>f__AnonymousType3`8) [Get-MgPolic
   yRol...ssignment_List1], RestException`1
    + FullyQualifiedErrorId : AadPremiumLicenseRequired,Microsoft.Graph.PowerShell.Cmdlets.GetMgPolicyRoleManagementPo
   licyAssignment_List1
Cannot bind argument to parameter 'UnifiedRoleManagementPolicyId' because it is an empty string.

#### Suggested solution to the issue
Fail gracefully if cmdlet throws an error

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
# insert configuration here
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2016 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 14393.5246.amd64fre.rs1_release.220701-1744
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of the DSC module that was used ('dev' if using current dev branch)
'dev' (1.22.824.1)

### LINKED ISSUE #2283: AADRoleSetting: ActivateApprover expects User eventhough Group also posible
#### Details of the scenario you tried and the problem that is occurring
Call Export-M365DSCConfiguration -Components @("AADRoleSetting") for a Role that has a Group set for Approvers
#### Verbose logs showing the problem

Connecting to {MicrosoftGraph}...✅
[1/1] Extracting [AADRoleSetting]...
    |---[1/94] Global AdministratorGet-MgUser : Resource '58a7761f-0793-4088-a63c-d41ca1b3bfee' does not exist or one of its queried reference-property
objects are not present.
At C:\Program
Files\WindowsPowerShell\Modules\Microsoft365DSC\DSCResources\MSFT_AADRoleSetting\MSFT_AADRoleSetting.psm1:230 char:9
+         $user = Get-MgUser -UserId $Item
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: ({ UserId = 58a7...ndProperty =  }:<>f__AnonymousType7`3) [Get-MgUser_
   Get], RestException`1
    + FullyQualifiedErrorId : Request_ResourceNotFound,Microsoft.Graph.PowerShell.Cmdlets.GetMgUser_Get

#### Suggested solution to the issue
Extend Get-TargetResource to also try Get-MgGroup if no valid User found
Extend Set-Target Resource with possibility to use Group Displayname/ObjectID 
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
# insert configuration here
```

#### Version of the DSC module that was used ('dev' if using current dev branch)
'dev'

### PR DESCRIPTION
**Pull request by sandrola**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # Change log for Microsoft365DSC
 
 # UNRELEASED
-
+* AADPIMRoleSetting
```
Comment (NikCharlebois) [inline CHANGELOG.md:4]: Should say AADRoleSetting

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADRoleSetting/MSFT_AADRoleSetting.psm1 [hunk 2]
```diff
@@ -1253,7 +1312,6 @@ function Export-TargetResource
                 ApplicationId         = $ApplicationId
                 TenantId              = $TenantId
                 CertificateThumbprint = $CertificateThumbprint
-                Credential            = $Credential
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADRoleSetting/MSFT_AADRoleSetting.psm1:1256]: Why are we removing credentials from here?

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADRoleSetting/MSFT_AADRoleSetting.psm1 (hunk 2 above)
Comment (sandrola) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADRoleSetting/MSFT_AADRoleSetting.psm1]: I think this was a mistake, i have corrected it in the next commit, Thanks.

**Review by NikCharlebois — APPROVED** [review]

**Comment by NikCharlebois** [comment]
Can you please resolved the conflicts by merging the latest changes and I'll your PR after? Thanks

**Comment by sandrola** [comment]
Hello Nik

I have updated my Branch (discarding the commits as proposed on GitHub), resolved the conflict on my machine and committed again. But now the Pull request is not there anymore. Is already merged?

Regards
Sandro

Sandro Lanfranchi
System Engineer
T 058 223 51 32, M 079 636 12 21
***@***.******@***.***>

Swisscom (Schweiz) AG
Microsoft Engineering & Operation
Pfingstweidstrasse 51, 8005 Zürich
www.swisscom.ch<https://www.swisscom.ch/>

Von: Nik Charlebois ***@***.***>
Gesendet: Montag, 19. September 2022 18:40
An: microsoft/Microsoft365DSC ***@***.***>
Cc: Lanfranchi Sandro, B2B-ITL-PWU-CVM-WHM ***@***.***>; Author ***@***.***>
Betreff: Re: [microsoft/Microsoft365DSC] Aad role setting (PR #2298)


Can you please resolved the conflicts by merging the latest changes and I'll your PR after? Thanks

-

**Pull request closed without merging**